namer/plural_namer.go

```diff
@@ -56,13 +58,61 @@ func (r *pluralNamer) Name(t *types.Type) string {
 	if plural, ok = r.exceptions[singular]; ok {
 		return r.finalize(plural)
 	}
-	switch string(singular[len(singular)-1]) {
-	case "s", "x":
-		plural = singular + "es"
-	case "y":
-		plural = singular[:len(singular)-1] + "ies"
+	if len(singular) < 2 {
+		return r.finalize(plural)
+	}
+
+	switch rune(singular[len(singular)-1]) {
+	case 's', 'x', 'z':
+		plural = esPlural(singular)
+	case 'y':
+		sl := rune(singular[len(singular)-2])
+		if isConsonant(sl) {
+			plural = iesPlural(singular)
+		} else {
+			plural = sPlural(singular)
+		}
+	case 'h':
+		sl := rune(singular[len(singular)-2])
+		if sl == 'c' || sl == 's' {
+			plural = esPlural(singular)
+		} else {
+			plural = sPlural(singular)
+		}
+	case 'n':
+		sl := rune(singular[len(singular)-2])
+		if sl == 'o' {
+			plural = aPlural(singular)
+		} else {
+			plural = sPlural(singular)
+		}
 	default:
-		plural = singular + "s"
+		plural = sPlural(singular)
 	}
 	return r.finalize(plural)
 }
+
+func iesPlural(singular string) string {
+	return singular[:len(singular)-1] + "ies"
+}
+
+func aPlural(singular string) string {
```

Comment:
I was thinking the same thing when I wrote it. The example noun in tests (`Phenomenon`) seems like a cool thing for CRD name but I don't feel strongly about keeping it.

Added the `lf -> ives` case. I originally left it out because it has fair amount of special cases but covering majority right seems better than failing on all (?)

Reply:
When in doubt, leave it out?

Reply:
Agree, removed